server/api.js

```diff
@@ -134,4 +135,64 @@ router.get("/tenders", async (req, res) => {
 	}
 });
 
+router.get("/tender/:id", async (req, res) => {
+	const tenderID = parseInt(req.params.id);
+	let page = parseInt(req.query.page) || 1;
+	const itemsPerPage = 10;
+	const totalBids = await db.query(
+		"SELECT COUNT(tender_id) FROM bid WHERE tender_id = $1",
+		[tenderID]
+	);
+	const totalPages = Math.ceil(totalBids.rows[0].count / itemsPerPage);
+	const offset = (page - 1) * itemsPerPage;
+
+	const result = await db.query(
+		"SELECT * FROM bid WHERE tender_id = $1 LIMIT $2 OFFSET $3",
+		[tenderID, itemsPerPage, offset]
+	);
+	result
+		? res.send({
+				results: result.rows,
+				pagination: {
+					itemsPerPage: 10,
+					currentPage: page,
+					totalPages: totalPages,
+				},
+		  })
+		: res.status(500).send({ code: "SERVER_ERROR" });
+});
+
+router.post("/bid-status/:bidId/:tenderId", async (req, res) => {
```

Comment:
Change the endpoint to `/bid/:bidId/status`. You don't need the tender id, you can get it based on the bid id